### PR TITLE
ref(suggested-solution): Add new design

### LINKED ISSUE #46584: UI & UX polish required for AI suggestion project
### Problem
The current implementation breaks Sentry's UI conventions

### Acceptance Criteria
- The user sees a permanent banner encouraging them to look at the suggestion
- The user is required to scroll below the breadcrumbs to find the AI suggestion Banner. Meaning, remove the header button.
- If the user has given their consent, we show AI-generated text in the format of, Problem, Solution & Joke
- The user can decide to rate their experience by clicking one of three buttons
- The user can collapse the section

### Designs

- [Figma](https://www.figma.com/file/8uOQuURhMQgRC72c81zOF4/Specs%3A-AI-Suggestions?node-id=0%3A1&t=ARHrvJHKoQ1mRVFv-1)
- [Art Assets](https://drive.google.com/file/d/1Fp5YRL4h6ORGcwuuv3W_5eSjUh79d02x/view?usp=sharing)

#### Banner Closed
Image

#### AI Suggestions Loading
Image

#### AI Suggestions Shown
Image

### PR DESCRIPTION
**Pull request by priscilawebdev**
Add [a new design](https://www.figma.com/file/8uOQuURhMQgRC72c81zOF4/Specs%3A-AI-Suggestions?node-id=5-587&t=QxYs5joK0T5HHnOU-0) (sorry internal only) for the suggested fix feature (currently only early-adopters).

**Previews:**

https://user-images.githubusercontent.com/29228205/231523309-6fdbe2ba-a72c-4c10-a987-2c1a995835d4.mov

Screenshot 2023-04-12 at 17 43 40
Screenshot 2023-04-12 at 17 44 32
Screenshot 2023-04-12 at 17 44 32

closes https://github.com/getsentry/sentry/issues/46584

**Remaining Tasks:**

- Add the feature flag `open-ai-suggestion-new-design` that is going to be controlled by flagr 
- Remove old implementation
- Update the backend to return other headings and message format
- Add Tests
- Fix hacks and todos described in the code 
- Update Docs with new copies and screenshots
- Create Analytics Dashboard

### CODE REVIEW

File: static/app/components/events/aiSuggestedSolution/suggestion.tsx [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    </Wrapper>
+  );
+}
+
+const Wrapper = styled(Panel)<{darkMode: boolean}>`
+  ${p =>
+    p.darkMode &&
+    css`
+      /* hack until we update the background of the loading video */
+      background: #2b2031;
```
Comment (priscilawebdev) [inline static/app/components/events/aiSuggestedSolution/suggestion.tsx:264]: this is for the loading that has a different background color

File: static/app/components/events/aiSuggestedSolution/suggestion.tsx [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+  padding: ${space(4)} 0;
+  text-align: center;
+  gap: ${space(2)};
+  display: flex;
+  flex-direction: column;
+`;
+
+const Content = styled('div')`
+  padding: ${space(2)};
+  /* hack until we update backend to send us other heading */
```
Comment (priscilawebdev) [inline static/app/components/events/aiSuggestedSolution/suggestion.tsx:309]: we will update the backend

File: static/app/utils/theme.tsx [hunk 3]
```diff
@@ -484,7 +484,7 @@ const generateBadgeTheme = (colors: BaseColors) => ({
     color: colors.white,
   },
   experimental: {
-    background: `linear-gradient(90deg, ${colors.blue300}, ${colors.green300})`,
+    background: `linear-gradient(270deg, #F5BE50 0%, #E06581 100%)`,
```
Comment (priscilawebdev) [inline static/app/utils/theme.tsx:487]: these colors are not part of our theme...shall we add them @Jesse-Box ?

File: static/app/components/events/aiSuggestedSolution/experimentalFeatureBadge.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      type="experimental"
+      title={t(
+        'This is an OpenAI generated solution that suggests a fix for this issue. Be aware that this may not be accurate.'
+      )}
+    />
+  );
+}
+
+const CapitalizesFeatureBadge = styled(FeatureBadge)`
+  text-transform: capitalize;
```
Comment (evanpurkhiser) [inline static/app/components/events/aiSuggestedSolution/experimentalFeatureBadge.tsx:19]: Do we really have to do this? We don't do this anywhere else?

Comment (priscilawebdev) [reply static/app/components/events/aiSuggestedSolution/experimentalFeatureBadge.tsx:19]: I understand your concern and noticed the same, but everything uppercase seems so aggressive 🤔 in any case for the sake of consistency, I agree and think we should remove this change... are you fine with uppercasing the tag @Jesse-Box ?

File: static/app/utils/theme.tsx (hunk 3 above)
Comment (Jesse-Box) [reply static/app/utils/theme.tsx:487]: Nah I would just use whatever badge variant you used previously.

Comment (priscilawebdev) [reply static/app/utils/theme.tsx:487]: oh but these colors are so cool! :sad

File: static/app/components/events/aiSuggestedSolution/experimentalFeatureBadge.tsx (hunk 4 above)
Comment (Jesse-Box) [reply static/app/components/events/aiSuggestedSolution/experimentalFeatureBadge.tsx:19]: I'm fine with using existing patterns here

File: static/app/utils/theme.tsx (hunk 3 above)
Comment (Jesse-Box) [reply static/app/utils/theme.tsx:487]: tbh, I just don't think it's important enough to extend the theme.

**Review by matejminar — APPROVED** [review]
I don't know where we landed on the mp4 vs animated svg loader but approving to unblock you and let's fix that once we have a clear decision.

File: static/app/components/events/aiSuggestedSolution/banner-background.svg
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/events/aiSuggestedSolution/banner-background.svg:1]: hm, thanks for the tip....will move them there

File: static/app/components/featureBadge.tsx [hunk 5]
```diff
@@ -60,9 +62,16 @@ function BaseFeatureBadge({
 
   return (
     <div {...props}>
-      <Tooltip title={title ?? defaultTitles[type]} disabled={noTooltip} position="right">
+      <Tooltip
+        title={title ?? defaultTitles[type]}
+        disabled={noTooltip}
+        position="right"
+        containerDisplayMode="inline-flex"
```
[the remark this answers is not in the document]
Comment (priscilawebdev) [reply static/app/components/featureBadge.tsx:69]: I had the same concern but the code wrapping this component is already using inline-flex and my hunch is that is trying to fix the same thing... sad that we don't use visual snapshots anymore to see if I regressed elsewhere... let us be on the saver side and remove it

**Comment by priscilawebdev** [comment]
thanks for the review @matejminar ... I plan to replace the mp4 with a new gif for now and after i will pair with @evanpurkhiser on animating the SVG

**Pull request merged**